pkg/cassandra/config/config.go

```diff
@@ -5,48 +5,44 @@
 	gocqlw "github.com/jaegertracing/jaeger/pkg/cassandra/gocql"
-	"github.com/jaegertracing/jaeger/pkg/config/tlscfg"
 )
 
-// Configuration describes the configuration properties needed to connect to a Cassandra cluster
+// Configuration describes the configuration properties needed to connect to a Cassandra cluster.
 type Configuration struct {
-	Servers              []string       `valid:"required,url" mapstructure:"servers"`
-	Keyspace             string         `mapstructure:"keyspace"`
-	LocalDC              string         `mapstructure:"local_dc"`
-	ConnectionsPerHost   int            `mapstructure:"connections_per_host"`
-	Timeout              time.Duration  `mapstructure:"-"`
-	ConnectTimeout       time.Duration  `mapstructure:"connection_timeout"`
-	ReconnectInterval    time.Duration  `mapstructure:"reconnect_interval"`
-	SocketKeepAlive      time.Duration  `mapstructure:"socket_keep_alive"`
-	MaxRetryAttempts     int            `mapstructure:"max_retry_attempts"`
-	ProtoVersion         int            `mapstructure:"proto_version"`
-	Consistency          string         `mapstructure:"consistency"`
-	DisableCompression   bool           `mapstructure:"disable_compression"`
-	Port                 int            `mapstructure:"port"`
-	Authenticator        Authenticator  `mapstructure:",squash"`
-	DisableAutoDiscovery bool           `mapstructure:"-"`
-	TLS                  tlscfg.Options `mapstructure:"tls"`
+	Schema     Schema     `mapstructure:"schema"`
+	Connection Connection `mapstructure:"connection"`
 }
 
-func DefaultConfiguration() Configuration {
-	return Configuration{
-		Servers:            []string{"127.0.0.1"},
-		Port:               9042,
-		MaxRetryAttempts:   3,
-		Keyspace:           "jaeger_v1_test",
-		ProtoVersion:       4,
-		ConnectionsPerHost: 2,
-		ReconnectInterval:  60 * time.Second,
-	}
+type Connection struct {
+	Servers              []string               `mapstructure:"servers" valid:"required,url" `
+	LocalDC              string                 `mapstructure:"local_dc"`
+	Port                 int                    `mapstructure:"port"`
+	DisableAutoDiscovery bool                   `mapstructure:"disable_auto_discovery"`
+	ConnectionsPerHost   int                    `mapstructure:"connections_per_host"`
+	ReconnectInterval    time.Duration          `mapstructure:"reconnect_interval"`
+	SocketKeepAlive      time.Duration          `mapstructure:"socket_keep_alive"`
+	MaxRetryAttempts     int                    `mapstructure:"max_retry_attempts"`
+	TLS                  configtls.ClientConfig `mapstructure:"tls"`
+	QueryTimeout         time.Duration          `mapstructure:"query_timeout"`
+	ConnectTimeout       time.Duration          `mapstructure:"connection_timeout"`
+	Authenticator        Authenticator          `mapstructure:"auth"`
+	ProtoVersion         int                    `mapstructure:"proto_version"`
+	Consistency          string                 `mapstructure:"consistency"`
```

Comment:
I'm starting to think you may want to partition these further into connection and query - it seems the fields are pretty separate for those.

Reply:
Created a separate grouping for query. Should `Consistency` be under `Query` as well?

Reply:
I think so - it's passed as part of queries, not as connection parameter, is it not?